### PR TITLE
Plain text parser

### PR DESCRIPTION
**Pull request by lucasmlofaro**
- Add parser for `txt` files
- `read()` just decodes content
- `create()` tries to cast the object as bytes and allows an `encoding` keyword argument

### CODE REVIEW

File: cabinets/parser/txt.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+@register_extensions('txt')
+class PlainTextParser(Parser):
+
+    @classmethod
+    def load_content(cls, content):
+        return content.decode()
+
+    @classmethod
+    def dump_content(cls, data, encoding='utf-8'):
```
Comment (samhollenbach) [inline cabinets/parser/txt.py:12]: The interesting thing about using a kwarg here is that there is no way to actually access it from standard `cabinets` methods (`cabinets.read()`, `cabinets.create()`). This brings up an interesting question of whether we need to allow for some configuration for parsers to be passed with the top level `cabinets` methods.

File: test/test_cabinet.py [hunk 2]
```diff
@@ -65,6 +65,25 @@ def test_read_create_pickle(self):
         result = cabinets.read(f'{protocol}://{filename}')
         self.assertDictEqual(data, result)
 
+    def test_read_plain_text(self):
+        protocol = 'file'
+        filename = os.path.join(self.fixture_path, 'sample.txt')
+        data = cabinets.read(f'{protocol}://{filename}')
+        expected = "I am sample text!\nThis file has more than one line.\n" \
```
Comment (samhollenbach) [inline test/test_cabinet.py:72]: May want to add some more special cases to this test, e.g. unicode characters?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
+illustrator/
```
Comment (samhollenbach) [inline .gitignore:1]: lol

File: cabinets/parser/txt.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from cabinets.parser import register_extensions, Parser
+
+
+@register_extensions('txt')
+class PlainTextParser(Parser):
+
+    @classmethod
+    def load_content(cls, content):
```
Comment (samhollenbach) [inline cabinets/parser/txt.py:8]: If `dump_content` has an `encoding` kwarg, should `load_content` not have one also? `bytes.decode()` does take an `encoding` keyword arg as well.

**Review by samhollenbach — COMMENTED** [review]
Few things to think about.

File: cabinets/parser/txt.py (hunk 1 above)
Comment (lucasmlofaro) [reply cabinets/parser/txt.py:12]: Yup, sounds like a problem that needs more thought. I'm just gonna default to `utf-8` and we should reevaluate in a new PR.

File: test/test_cabinet.py (hunk 2 above)
Comment (lucasmlofaro) [reply test/test_cabinet.py:72]: Yeah, I just added some non-ascii characters, should be good enough.

File: cabinets/parser/txt.py (hunk 4 above)
Comment (lucasmlofaro) [reply cabinets/parser/txt.py]: I was expecting this lol.

**Review by samhollenbach — APPROVED** [review]

**Pull request merged**